fineract-provider/src/main/java/org/apache/fineract/portfolio/delinquency/service/DelinquencyWritePlatformServiceImpl.java

```diff
@@ -134,8 +144,61 @@ public CommandProcessingResult deleteDelinquencyBucket(Long delinquencyBucketId,
                         "Data integrity issue with resource: " + delinquencyBucket.getId());
             }
             repositoryBucket.delete(delinquencyBucket);
+            return new CommandProcessingResultBuilder().withCommandId(command.commandId()).withEntityId(delinquencyBucket.getId()).build();
         }
-        return new CommandProcessingResultBuilder().withCommandId(command.commandId()).withEntityId(delinquencyBucket.getId()).build();
+        return new CommandProcessingResultBuilder().withCommandId(command.commandId()).withEntityId(delinquencyBucketId).build();
+    }
+
+    @Override
+    public LoanScheduleDelinquencyData getOverdueDays(LoanScheduleDelinquencyData loanScheduleDelinquencyData) {
+        final Loan loan = this.loanRepository.findOneWithNotFoundDetection(loanScheduleDelinquencyData.getLoanId());
+        loanScheduleDelinquencyData.setLoan(loan);
+        loanScheduleDelinquencyData.setOverdueDays(getOverdueDays(loan));
+        return loanScheduleDelinquencyData;
+    }
+
+    @Override
+    public Long getOverdueDays(Loan loan) {
```

Comment:
Shouldn't we reuse a similar logic than in `LoanReadPlatformServiceImpl#retrieveAllOverdueInstallmentsForLoan`? Whats the logical difference from this?

Reply:
The idea is to have the overdue loan installments, one case is with the overdue direct and the second is througj tje loan transaction charge back